### PR TITLE
LPS-182347 Add asset Tag to the searching filter

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://issues.liferay.com/browse/LPS-182347

It's under FF LPS-84424

To test :

1. Go to Documents & Media
2. Create files with some tags
3. You can see the tags from the group
4. Select tag from modal
5. only the documents with this tag are shown

https://user-images.githubusercontent.com/47524751/234052654-b698cc3a-d085-48f8-8081-aee40588fa7a.mp4

- LPS-182347 add action to the modal select
- LPS-182347 put the tags on the label item List
- LPS-182347 select all on the dropdown when the tags are empty and deselect the tag when you select all
- LPS-182347 add the assetTagIds to the search content

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1317401_7819

**Comment by Hanlf** [comment]
The Poshi test failure is not related to this PR.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java [hunk 1]
```diff
@@ -626,6 +631,36 @@ public Boolean isShowSearch() {
...
+								curAssetTagId -> {
+									if (Objects.equals(
+											assetTagId, curAssetTagId)) {
+
+										return null;
+									}
+
+									return curAssetTagId;
+								},
+								String.class)));
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:657]: The current `_getRemoveLabelURL` method doesn't seem correct to me:
* What it currently does is just returning an URL with a parameter, not very related to "RemoveLabel"
* If you want to still use it I would hide the logic of `TransformUtil.transformToArray` inside this method
* Or I think is better to transform the set to an array and work like in `_addExtensionFilterLabelItems`, which uses method `ArrayUtil.remove()`

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java [hunk 2]
```diff
@@ -890,6 +925,12 @@ private long _getFolderId() {
 		return _dlAdminDisplayContext.getFolderId();
 	}
 
+	private String _getLabel(String key, String value) {
+		return StringBundler.concat(
+			LanguageUtil.get(httpServletRequest, key), StringPool.COLON,
+			StringPool.SPACE, value);
+	}
+
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:935]: Currently the label is built in several places with this code (yours is also OK):

```java
	String.format(
		"%s: %s",
		LanguageUtil.get(_httpServletRequest,  key),
		HtmlUtil.escape(value));
```

Creating a method for it seems OK. Please refactor all places to use the newly created method. And don't forget to add the call to `HtmlUtil.escape()`.

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]
Good work! Just some minor changes...

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:935]: I will use the `HtmlUtil.escape(`, thanks!

I have to think about 
`String.format("%s: %s"` vs `StringBundler.concat(`

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 1 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:657]: > The name of the method is what returns (removeLabelURL) , so the name on this case is correct 🤷🏻‍♀️

You can call the method `_getUrl` and would be still correct. That's my point.

> On _addExtensionFilterLabelItems you are removing an element of the array you are in (on the for) I prefer not to do it.
> Thats why I prefer the use of the TransformUtil.transformToArray( and my idea is to use it on the categories ticket.

`ArrayUtil.remove()` doesn't modify the input array.
Anyway, if you prefer to use `TransformUtil.transformToArray` I would do this work inside the method

Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:657]: If I reuse the method as it like with extension:

```
		_getRemoveLabelURL(
				"extension",
			() -> ArrayUtil.remove(extensions, extension)));
```
							
You are ok with it?

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
ci:test:relevant

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:forward

**Pull request closed without merging**